src/esp/gfx/RenderCamera.cpp

```diff
@@ -209,11 +210,20 @@ esp::geo::Ray RenderCamera::unproject(const Mn::Vector2i& viewportPosition) {
           Magnum::Vector2{1.0f},
       1.0};
 
+  // compute the far plane
+  auto zFar =
+      Mn::Frustum::fromMatrix(projectionMatrix() * cameraMatrix()).far();
+  float farDistance = zFar[3] / zFar.xyz().length();
+
   ray.direction =
       ((object().absoluteTransformationMatrix() * projectionMatrix().inverted())
            .transformPoint(normalizedPos) -
-       ray.origin)
-          .normalized();
+       ray.origin) /
+      farDistance;
+
+  if (normalized) {
+    ray.direction = ray.direction.normalized();
+  }
```

Comment:
(Again ignore if this utility isn't meant to be used in performance-sensitive scenarios.)

A better way would be to have this split into two functions, one `unproject()` that doesn't normalize, and an `unprojectNormalized()` that calls `unproject()`, normalizes its result and returns it. That way there's no branch. (Though that way it'll be a breaking change, because `unproject()` will not normalize anymore, so you might want to invent different naming of the two overloads.)

Reply:
Yeah, thanks. I thought about this also, but didn't want a breaking change or to add more bulk to the API. I think this one can stay for now.